From da1c7fd61d13a590ee9b0262d76c81db3753dda3 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Wed, 24 Feb 2016 14:37:12 +0800
Subject: [PATCH 2/3] yaffs2: NULL ->read/->write

As stated in commit 5d5d56897530 ("make new_sync_{read,write}() static"),
we shouldn't use new_sync_{read_write} for ->read/->write. We should
NULL ->read/->write with non-NULL ->{read,write}_iter will do the
right thing.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 fs/yaffs2/yaffs_vfs.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/fs/yaffs2/yaffs_vfs.c b/fs/yaffs2/yaffs_vfs.c
index fbbf31303537..0a3e9dabe1aa 100644
--- a/fs/yaffs2/yaffs_vfs.c
+++ b/fs/yaffs2/yaffs_vfs.c
@@ -794,8 +794,6 @@ static int yaffs_sync_object(struct file *file, struct dentry *dentry,
 
 #if (LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 22))
 static const struct file_operations yaffs_file_operations = {
-	.read = new_sync_read,
-	.write = new_sync_write,
 	.read_iter = generic_file_read_iter,
 	.write_iter = generic_file_write_iter,
 	.mmap = generic_file_mmap,
-- 
2.5.0

